source/core_http_client.c

```diff
@@ -1314,7 +1314,8 @@ static HTTPStatus_t addHeader( HTTPRequestHeaders_t * pRequestHeaders,
     size_t toAddLen = 0U;
     size_t backtrackHeaderLen = 0U;
 
-    /* These variables are here to pass into memcpy for MISRA compliance */
+    /* Directly passing in these strings to memcpy is a MISRA violation
```

Comment:
may want to mention which misra rule it violates.

Reply:
Good idea, added the rule number to the comment.